editor/project_settings_editor.cpp

```diff
@@ -1713,6 +1730,38 @@ void ProjectSettingsEditor::_editor_restart_close() {
 	restart_container->hide();
 }
 
+void ProjectSettingsEditor::_joypad_capture_timeout() {
+	if (add_type == INPUT_JOY_BUTTON) {
+		for (int i = 0; i < JOY_BUTTON_MAX; i++) {
+			if (Input::get_singleton()->is_joy_button_pressed(_get_current_device(), i)) {
+				device_index->select(i);
+				break;
+			}
+		}
+		return;
+	} else if (add_type == INPUT_JOY_MOTION) {
+		for (int i = 0; i < JOY_AXIS_MAX; i++) {
+			float axisValue = Input::get_singleton()->get_joy_axis(_get_current_device(), i);
+			if (ABS(axisValue - _last_axis_reading[i]) > 0.1f) {
+				if (axisValue < -0.75f) {
+					device_index->select(i * 2);
+				} else if (axisValue > 0.75f) {
+					device_index->select(i * 2 + 1);
+				}
```

Comment:
Avoid "magic numbers", use a constant to define the threshold:

```suggestion
				// Only choose the axis if it has been pushed almost all the way through.
				const float axis_threshold = 0.75;
				
				if (axisValue < -axis_threshold) {
					device_index->select(i * 2);
				} else if (axisValue > axis_threshold) {
					device_index->select(i * 2 + 1);
				}
```

Reply:
Sure. Meant to do that before I submitted it. Will ammend.